pkg/integrations/downloader/downloader.go

```diff
@@ -0,0 +1,137 @@
+// Package downloader implements a file downloader that can download files from
+// a URL and verify checksums of the file.
+package downloader
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sync"
+)
+
+// Global is a globally shared downloader.
+var Global = New(nil)
+
+type Downloader struct {
+	client *http.Client
+
+	// locks holds a lock for downloading a specific file.
+	lockMtx sync.Mutex
+	locks   map[string]*sync.Mutex
```

Comment:
This will leak memory if an unbounded list of files is downloaded, but that won't happen in our case. Maybe I should document that here in case someone imports this package.